### PR TITLE
Crop GeoArrays

### PR DESCRIPTION
**Pull request by KrOstir**
After adding the option to subset GeoArrays in #72 it is easy to implement #39.

Method to crop by coordinate box and by another GeoArray are implemented. 

```julia
function crop(ga::GeoArray, cbox::NamedTuple{(:min_x, :min_y, :max_x, :max_y)})
function crop(ga::GeoArray, cga::GeoArray)
```
The first one uses cropping box. Function `bbox_overlap` checks if it is correctly built, and it overlaps the GeoArray. The second method checks if both GeoArrays use the same CRS and then calls the first one with the `bbox` of the second one.

crop! is not implemented yet. We have to think if modifying the GeoArray inplace is required/desired.

In #39 @evetion mentions

> For the last signature, it can only work for GeoArrays with the same CRS. It will throw a warning if one of the rasters has a different affine transform (other resolution/rotation), because the cropping will introduce errors.

The first part is implemented. The second one should not be necessary, as we are cropping only one GeoArray, and we are using its resolution, rotation. The other one provides only the box.

### CODE REVIEW

File: src/geoutils.jl [hunk 1]
```diff
@@ -111,3 +111,73 @@ function flipud!(ga::GeoArray)
...
+        error("GeoArray and crop box do not overlap")
+    end
+    # Check extent and get bbox indices
+    ga_x, ga_y, = size(ga)
+    i_min_x, i_min_y = indices(ga, [cbox.min_x, cbox.max_y])
+    i_max_x, i_max_y = indices(ga, [cbox.max_x, cbox.min_y])
+
+    # bbox is larger
+    i_max_x -= 1
+    i_max_y -= 1
```
Comment (evetion) [inline src/geoutils.jl:163]: This shouldn't be required? See #79 for a breaking change for these conversions such as `indices`, which now use the center coordinates.

File: src/geoutils.jl [hunk 2]
```diff
@@ -111,3 +111,73 @@ function flipud!(ga::GeoArray)
...
+
+    # Determine indices for crop area
+    i_min_x = max(i_min_x, 1)
+    i_max_x = min(i_max_x, ga_x)
+    i_min_y = max(i_min_y, 1)
+    i_max_y = min(i_max_y, ga_y)
+
+    # Subset and return GeoArray
+    return ga[i_min_x:i_max_x,i_min_y:i_max_y,begin:end]
```
Comment (evetion) [inline src/geoutils.jl:172]: `begin:end` is the same as `:`

**Review by evetion — CHANGES_REQUESTED** [review]
I like it! Some small comments and note that an older commit (README updates) is in this PR, while already merged.

File: src/geoutils.jl (hunk 1 above)
Comment (KrOstir) [reply src/geoutils.jl:163]: I like what you did in #79. This can be removed when it is merged to master.

File: src/geoutils.jl (hunk 2 above)
Comment (KrOstir) [reply src/geoutils.jl:172]: I am at Julia begin :-)

**Comment by KrOstir** [comment]
I hope README was successfully removed from PR.

**Comment by evetion** [comment]
Thx for the PR!

**Comment by KrOstir** [comment]
Happy to contribute.

**Pull request merged**